pkg/kv/kvserver/rditer/replica_data_iter_test.go

```diff
@@ -283,27 +276,46 @@ func TestReplicaDataIterator(t *testing.T) {
 		parName := fmt.Sprintf("r%d", tc.desc.RangeID)
 		t.Run(parName, func(t *testing.T) {
 			testutils.RunTrueAndFalse(t, "replicatedOnly", func(t *testing.T, replicatedOnly bool) {
-				replicatedSpans := []ReplicatedSpansFilter{ReplicatedSpansAll, ReplicatedSpansExcludeUser, ReplicatedSpansUserOnly}
-				for i := range replicatedSpans {
-					replicatedKeysName := "all"
-					switch replicatedSpans[i] {
-					case ReplicatedSpansExcludeUser:
-						replicatedKeysName = "exclude-user"
-					case ReplicatedSpansUserOnly:
-						replicatedKeysName = "user-only"
-					}
-					t.Run(fmt.Sprintf("replicatedSpans=%v", replicatedKeysName), func(t *testing.T) {
-						name := "all"
+				// Test all combinations of boolean fields.
```

Comment:
Why do we need the combinatorics in a datadriven test? Reviewing all these files is infeasible and nobody will do.

I think what's useful is to have one "select all" test, and/or one test which does mini-select for each of the possible sub-spans. This serves as a documentation + golden test.

The combinatorics of things can then be tested with a regular unit-test. We want 2 properties: `select(A|B) == select(A)|select(B)`, and `select(anything)` is sorted by key. These kinds of properties/invariants are better to check in code than by eyeballing.

Reply:
Demoed this on `TestSelect` in the `rditer: improve TestSelect` commit.